packages/core/src/utils.ts

```diff
@@ -164,10 +164,7 @@ export const getGasPriceOverrides = async (
     throw new Error(`Unable to retrieve latest block.`)
   }
 
-  if (
-    executionMode === ExecutionMode.LocalNetworkCLI ||
-    executionMode === ExecutionMode.Platform
```

Comment:
Just noticed this bug when testing with the website. We cannot use `ExecutionMode.Platform` b/c it will result in the gas limit being extremely high for the production executor, which will cause transactions to fail due to insufficient funds. 

I'm not sure if there was a specific reason why you included this, though, so let me know if you think some additional changes need to be made related to this.

Reply:
I'd like to avoid regressions like this, and I think there's a straightforward solution so I've added an environment variable `SPHINX_INTERNAL__HARDCODE_GAS_LIMIT`. It can be used to enable this optimization. I recommend setting it explicitly in test files. I've done so in both `artifacts.spec.ts` and `propose.spec.ts`.

Reply:
I don't think that resolves the issue because we need the optimization enabled in the *production* proposal process (i.e. during the user's proposal). The user's proposal process is really slow without the optimization. It seems like your solution would only enable the optimization when we're testing the proposal process.

Two options come to mind:
1. The simplest solution may be to use the original if-statement that existed before this PR, and add an env variable that the executor can specify to *disable* the optimization. (Instead of requiring an environment variable to *enable* the optimization everywhere else).
2. I think the proper but more involved solution is to add another `ExecutionMode` for the production executor. I think this may be a more robust solution because otherwise we're operating under the assumption that the proposal simulation and the production executor share all of the same logic (except for this single if-statement, which would be resolved via env variable). Fwiw, when I was writing the logic for the `Platform` execution mode, I was only thinking about the proposal simulation and not the production executor. There could be other errors that arise because of this oversight on my part

I think option 1 is a reasonable solution for this PR given that you've recently tested that the executor works. If you choose option 1, one of us should make a ticket for option 2. Again, ping me on Telegram if you want me to make that ticket